internal/cli/create_cmd.go

```diff
@@ -0,0 +1,147 @@
+package cli
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/tilt-dev/tilt/internal/analytics"
+	engineanalytics "github.com/tilt-dev/tilt/internal/engine/analytics"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+// A human-friendly CLI for creating cmds.
+//
+// The name is unfortunate.
+//
+// (as opposed to the machine-friendly CLIs of create -f or apply -f)
+type createCmdCmd struct {
+	helper *createHelper
+
+	dir         string
+	env         []string
+	filewatches []string
+}
+
+var _ tiltCmd = &createCmdCmd{}
+
+func newCreateCmdCmd() *createCmdCmd {
+	helper := newCreateHelper()
+	return &createCmdCmd{
+		helper: helper,
+	}
+}
+
+func (c *createCmdCmd) name() model.TiltSubcommand { return "cmd" }
+
+func (c *createCmdCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "cmd NAME COMMAND [ARG...]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Create a local command in a running tilt session",
+		Long: `Create a local command in a running tilt session.
+
+Intended to compose with other Tilt APIs that
+can restart the command or monitor its status.
+
+COMMAND should be an executable. A shell script will not work.
+
+To run a shell script, use 'sh -c' (as shown in the examples).
+`,
+		Args: cobra.MinimumNArgs(2),
+		Example: `
+tilt alpha create cmd my-cmd echo hello world
+
+tilt alpha create cmd my-cmd sh -c "echo hi && echo bye"
+`,
+	}
+
+	// Interpret any flags after the object name as part of the
+	// command args.
+	cmd.Flags().SetInterspersed(false)
+
+	cmd.Flags().StringVarP(&c.dir, "workdir", "w", "",
+		"Working directory of the command. If not specified, uses the current working directory.")
+	cmd.Flags().StringArrayVarP(&c.env, "env", "e", nil,
+		"Set environment variables in the form NAME=VALUE.")
+	cmd.Flags().StringSliceVar(&c.filewatches, "file-watch", nil,
```

Comment:
This is a total bikeshed and not important, but since the command + API convention is just `filewatch` (no hyphen), it might make sense to carry that here

Reply:
i agree! changed